### DISCUSSION TITLE
Been with Github a year

### DISCUSSION THREAD

**Discussion by Anthonymcqueen21** [community/community#21559] [New to GitHub]
<p>Hello fellow programmers i have been with Github for a year now and i feel i barely scratched the surface on what this site can do and offer. I made a few repositories showcasing my work general files and contributed on a few repositories but i want more.</p>
<p>I have been told that Git is a very usfull tool for programmers can someone explain how to use Git and what can i contribute towards it. This may be a generic question but i would like the details if possible fellow members who are more experienced in the developement world.</p>
<p>Thank you.</p>

**Comment by chrisparnin** [comment, marked as answer]
<p>For someone learning more advanced features of git, I always found this interactive tutorial an interesting way to play around with the features. I use this in my university courses and find that students always respond well to it:</p>
<p><strong>Git Branching Playground</strong></p>
<p>We will solve the “Introduction Sequence” levels in:</p>
<p><a href="http://pcottle.github.io/learnGitBranching/" rel="nofollow noopener">http://pcottle.github.io/learnGitBranching/</a></p>

**Comment by Vic-ST** [comment]
<p>I recommend you take <a href="https://www.codecademy.com/learn/learn-git" title="Git" rel="nofollow noopener">this</a> course on git by Codecademy. That is how I started out programming git.</p>

**Comment by PN1019** [comment]
<p>Few interesting course, ebook and articles I would recommend here : </p>
<p><a href="https://opensource.com/article/17/6/github-driven-development?utm_campaign=intrel" title="A guide to open source your bootstrap project wirh github" rel="nofollow noopener">https://opensource.com/article/17/6/github-driven-development?utm_campaign=intrel</a></p>
<p><a href="https://opensource.com/life/16/8/how-construct-your-own-git-server-part-6?utm_campaign=intrel" title="How to construct your own git server" rel="nofollow noopener">https://opensource.com/life/16/8/how-construct-your-own-git-server-part-6?utm_campaign=intrel</a></p>
<p><a href="https://git-scm.com/book/en/v2" title="ebook" rel="nofollow noopener">https://git-scm.com/book/en/v2</a></p>
<p><a href="https://opensource.guide/" title="Opensource guide" rel="nofollow noopener">https://opensource.guide/</a></p>
<p><a href="https://www.udacity.com/course/how-to-use-git-and-github--ud775" title="Udacity course link" rel="nofollow noopener">https://www.udacity.com/course/how-to-use-git-and-github–ud775</a></p>
<p>Hope this helps!!!</p>

**Comment by Anthonymcqueen21** [comment]
<p>Thank you i will check it out and see what i can do.</p>

**Comment by Anthonymcqueen21** [comment]
<p>I have not been on codeacademy in a long time since i started programming i will refer back to the site and learn the Git.</p>
<p>Thank you.</p>
